esx_service/utils/auth.py

```diff
@@ -91,34 +91,34 @@ def get_default_tenant():
         return None, None, None
 
 
-def get_default_privileges():
-    """ Get DEFAULT privileges by querying the auth DB.
-        DEFAULT privilege will match any (tenant_uuid, datastore) pair which
-        does not have an entry in privileges table explicitly
+def get_all_ds_privileges(tenant_uuid):
+    """ Get privileges on _ALL_DS by querying the auth DB.
+        For the tenant with given tenant_uuid, this privilege
+        will match any datastore which does not have an entry
+        in privileges table explicitly
```

Comment:
@lipingxue, ok, so if the DB has <vmgroup-uuid, _ALL_DS_> it will match any datastore. But then will it allow creating a volume on the datastore? Say we have VMG1 and VMG2 with DS1 and DS2 as default datastores. And say VMG1 has the __ALL_DS_ specified in the privileges table, The user may issue "volume create x@DS2 from VMG1 and the volume is created on DS2. Is that correct?

May be ok, but vmgroups are to ensure user volumes are isolated from each other via vmgroups. In which case this catch-all shouldn't be allowed.